### PR TITLE
feat(contracts): deploy factory via create2 [SBK-211]

### PR DESCRIPTION
**Pull request by fubuloubu**
Had to add method to deploy using https://github.com/pcaversaccio/create2deployer for both the StreamManager blueprint and the factory contract

~~Currently debuging why it doesn't work properly to deploy with~~

### CODE REVIEW

**Comment by fubuloubu** [comment]
Deployed `StreamFactory` to 0x92823EB2DB42b8df354EDB5A1FB3668057e2935D

**Comment by fubuloubu** [comment]
Deployed `StreamManager` via `StreamFactory` to 0x3543Faeeddb7bAbCbBB216B3627f9c5E0C39CE41

File: scripts/deploy.py [hunk 1]
```diff
@@ -66,11 +102,31 @@ def manager(account, network, owner, min_stream_life, validators, tokens):
...
+
+        tx = factory.create(list(validators), token_addresses, sender=account)
+        try:
+            manager = project.StreamManager.at(tx.return_value)
+        except ApeException:
+            manager = project.StreamManager.at(factory.deployments(account))
+
+        cli_ctx.logger.success(f"StreamManager deployed to '{manager.address}'.")
+
+        if click.confirm("Publish"):
```
Comment (antazoey) [inline scripts/deploy.py:120]: for any click confirm, it would be nice to have flags for skipping.

File: scripts/deploy.py [hunk 2]
```diff
@@ -21,38 +23,72 @@ def cli():
...
-                + len_bytes
-                + b"\x3d\x81\x60\x0a\x3d\x39\xf3"
-                + blueprint_bytecode
+@click.option("--create2", default=None, help="A string tag for the create2 deployment salt")
+def factory(cli_ctx, account, network, blueprint, create2):
+    if create2:
+        # NOTE: This is the deployment address listed on the create2 deployer's github:
+        # https://github.com/pcaversaccio/create2deployer/tree/main#deployments-create2deployer
+        # TODO: Add SDK for create2 deployer
+        create2_deployer = Contract("0x13b0D85CcB8bf860b6b79AF3029fCA081AE9beF2")
```
Comment (antazoey) [inline scripts/deploy.py:32]: I wonder if deploying should be part of the SDK too so it is more testable

**Review by antazoey — APPROVED** [review]

File: scripts/deploy.py (hunk 2 above)
Comment (fubuloubu) [reply scripts/deploy.py:32]: My idea was that the `manager` address would be the transition point between the scripts and the SDKs. They all reference a specific deployment, those contracts weren't build for add-hoc deployments. Could look into it further down the line though, especially using the factory's `create` method properly

**Pull request merged**